### PR TITLE
Docs: clean up / reorganize Infrastructure documentation

### PR DESCRIPTION
**Pull request by angela-tran**
This PR  reorganizes the infrastructure documentation to be friendlier to a new developer.

- Added an intro section with more context on how to get access
- Moved up sections that explain the environments and architecture
- Made some content edits to remove unnecessary detail  and use clearer wording

### CODE REVIEW

**Comment by angela-tran** [comment]
@lalver1 @thekaveman I opened this new PR to replace #2129. This one has cleaner commits and is hopefully easier to review.

File: docs/deployment/infrastructure.md [hunk 1]
```diff
@@ -2,9 +2,76 @@
...
+- Networking
+- Front Door
+  - Web Application Firewall (WAF)
+  - Distributed denial-of-service (DDoS) protection
+- IAM
+- Service connections
+
+You'll see these referenced in Terraform as [data sources](https://developer.hashicorp.com/terraform/language/data-sources), meaning they are managed outside of Terraform.
+
+### Architecture
```
Comment (thekaveman) [inline docs/deployment/infrastructure.md:43]: I think this and many of the sections below can be brought up a level, i.e. this one to `##`.

In general, favor less depth unless there is a really strong connection between e.g. `##` and `###`. I would expect very few if any `####` level stuff.

This makes the table of contents easier to use and render more clearly.

**Review by thekaveman — CHANGES_REQUESTED** [review]
In general I really like this reorg of the content.

I think some of the sections are too deep though, see my comment below.

File: docs/deployment/infrastructure.md (hunk 1 above)
Comment (thekaveman) [reply docs/deployment/infrastructure.md:43]: E.g.

```
## Environments

### Ownership

## Architecture

## Naming conventions

## Making changes

## Azure environment setup
```

**Review by lalver1 — COMMENTED** [review]
This looks really good @angela-tran, and it's super useful for a new developer 👍

File: docs/deployment/infrastructure.md (hunk 1 above)
Comment (angela-tran) [reply docs/deployment/infrastructure.md]: No problem with those levels. Moved them in https://github.com/cal-itp/benefits/pull/2159/commits/f4d2d5b08c7462c598c1535a279f5fa7f9bb4bdc

**Review by thekaveman — APPROVED** [review]
:memo:

**Pull request merged**